code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -191,3 +198,363 @@
+			if(istype(C, /mob/living/simple_animal/hostile/headcrab))
+				continue
+			C.adjustFireLoss(10)
+			C.adjustToxLoss(10)
+			if(C.reagents)
+				C.reagents.add_reagent("sacid", 10)
+				to_chat(C, "<span class='userdanger'>You have been splashed with acid!</span>")
+
+/obj/effect/temp_visual/gonarch_artillery
+	icon = 'icons/mob/actions/actions.dmi'
+	icon_state = "sniper_zoom"
+	layer = BELOW_MOB_LAYER
+	light_range = 2
+	light_color = "#000000"
+	duration = 9
+
+/obj/effect/temp_visual/gonarch_artillery/ex_act()
+	return
+
+/obj/effect/temp_visual/gonarch_artillery/Initialize(mapload, list/flame_hit)
+	. = ..()
+	INVOKE_ASYNC(src, .proc/fall, flame_hit)
+
+/obj/effect/temp_visual/gonarch_artillery/proc/fall(list/flame_hit)
+	var/turf/T = get_turf(src)
+	playsound(T,'sound/magic/fleshtostone.ogg', 80, TRUE)
+	new /obj/effect/temp_visual/fireball(T)
+	sleep(duration)
+	playsound(T, "sound/effects/splat.ogg", 80, TRUE)
+	for(var/mob/living/L in T.contents)
+		if(istype(L, /mob/living/simple_animal/hostile/headcrab))
+			continue
+		if(L.reagents)
+			L.reagents.add_reagent("sacid", 20)
+			to_chat(L, "<span class='userdanger'>You have been splashed with acid!</span>")
+		L.adjustBruteLoss(35)
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/mode_switch(mode_to_switch, sound_to_play, clear_targets = TRUE,  change_icon)
+	if(mode_to_switch && mode_to_switch != mode)
+		mode = mode_to_switch
+		if(clear_targets)
+			LoseTarget()
+		if(sound_to_play)
+			var/sound_picked = pick(sound_to_play)
+			playsound(src, sound_picked, 200, TRUE, 15, pressure_affected = FALSE)
+		if(change_icon)
+			icon_state = change_icon
+			icon_living = change_icon
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rest()
+	if(get_area(src) == home_nest_area) // If I am home & Everything is calm
+		if(aggro_vision_range != gonarch_aggro_vision_range || vision_range != gonarch_standard_vision_range || speed != gonarch_standard_speed)
+			aggro_vision_range = gonarch_aggro_vision_range
+			vision_range = gonarch_standard_vision_range
+			speed = gonarch_standard_speed
+		if(homesick)
+			homesick = 0
+			step_towards(src, home_nest_turf, speed)
+		make_headcrabs() //Its its own proc, so they can be called by admins.
+	else
+		if(homesick >= 5)
+			mode_switch(GONARCH_MODE_RETURNING, gonarch_soundlist_return, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_return()
+	speed = -1
+	var/turf/olddist = get_dist(src, home_nest_turf)
+	step_towards(src, home_nest_turf, speed)
+	target = home_nest_turf
+	if(get_dist(src, target) >= olddist)
+		frustration++
+		if(frustration > frustration_limit)
+			DestroyPathToTarget()
+			homesick++
+	else
+		frustration = 0
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_newhome()
+	var/list/turfs = list()
+	var/list/turfs_in_range = range(4, src)
+	for(var/turf/T in range(gonarch_nestfind_range, src))
+		if(T in turfs_in_range)
```

Comment:
Best to just do a distance check here instead of this one. Distance checks are cheaper I believe than getting a range and then checking if an object is in said range